container_registry/container_analysis/samples.py

```diff
@@ -0,0 +1,343 @@
+        ValueError: If the parameters are invalid.
+    """
+    client = container_analysis_client.ContainerAnalysisClient()
+    note_name = client.note_path(project_id, note_id)
+    response = client.get_note(note_name)
+    return response
+# [END get_note]
+
+
+# [START get_occurrence]
+def get_occurrence(occurrence_name):
+    """Retrieves an occurrence based on it's name
+
+    :param occurrence_name: the name of the occurrence to delete.
+                in format "projects/{projectId}/occurrences/{occurrence_id}"
+    :return: the specified Occurrence object
+    :raises
+        google.api_core.exceptions.GoogleAPICallError: If the request
+            failed for any reason.
+        google.api_core.exceptions.RetryError: If the request failed due
+            to a retryable error and retry attempts failed.
+        ValueError: If the parameters are invalid.
+    """
+    client = container_analysis_client.ContainerAnalysisClient()
+    return client.get_occurrence(occurrence_name)
+# [END get_occurrence]
+
+
+# [START discovery_info]
+def get_discovery_info(image_url, project_id):
+    """prints the Discovery occurrence created for a specified image
+    This occurrence contains information about the initial scan on the image
+
+    :param image_url:  the Container Registry URL associated with the image
+                example: "https://gcr.io/project/image@sha256:foo"
+    :param project_id: the GCP project the occurrence will be created under
+    :raises
+        google.api_core.exceptions.GoogleAPICallError: If the request
+            failed for any reason.
+        google.api_core.exceptions.RetryError: If the request failed due
+            to a retryable error and retry attempts failed.
+        ValueError: If the parameters are invalid.
+    """
+    filterStr = "kind=\"DISCOVERY\" AND resourceUrl=\"" + image_url + "\""
+    client = container_analysis_client.ContainerAnalysisClient()
+    project_name = client.project_path(project_id)
+    response = client.list_occurrences(project_name, filter_=filterStr)
+    for occ in response:
+        print occ
+# [END discovery_info]
+
+
+# [START occurrences_for_note]
+def get_occurrences_for_note(note_id, project_id):
+    """Retrieves all the occurrences associated with a specified note
+
+    :param note_id: the note's unique identifier
+    :param project_id: the project's unique identifier
+    :return: number of occurrences found
+    :raises
+        google.api_core.exceptions.GoogleAPICallError: If the request
+            failed for any reason.
+        google.api_core.exceptions.RetryError: If the request failed due
+            to a retryable error and retry attempts failed.
+        ValueError: If the parameters are invalid.
+    """
+    client = container_analysis_client.ContainerAnalysisClient()
+    note_name = client.note_path(project_id, note_id)
+
+    response = client.list_note_occurrences(note_name)
+    count = 0
+    for o in response:
```

Comment:
Does `return len(response)` do the same?

Reply:
I considered it, but I thought it would be useful to demonstrate how to access the individual responses so the sample is easier to build real uses cases off of, and to keep it more consistent with other languages. Do you think it's better to keep it simple?

Reply:
If you want to leave in the loop, maybe add a comment 

    # you can process results here